source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -1099,7 +1120,7 @@ void Graph::setUiNodeInfo(UiNodePtr node, const std::string& type, const std::st
     {
         if (node->getNode())
         {
-            mx::NodeDefPtr nodeDef = node->getNode()->getNodeDef(node->getNode()->getName());
+            mx::NodeDefPtr nodeDef = node->getNodeDef();
```

Comment:
See include file where the nodedef is cached on UInode initialization. 
I simplified this as the cost of querying the nodedef is non-trivial and shows up in performance profiles.